src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -63,6 +64,321 @@ internal override string ExtensionName
+            {
+                if (isDefinition)
+                {
+                    var cciTypeParameter = (Cci.IGenericParameter)typeParameter.GetCciAdapter();
+                    Debug.Assert(cciTypeParameter.Variance == Cci.TypeParameterVariance.NonVariant);
+
+                    if (cciTypeParameter.MustBeReferenceType)
+                    {
+                        builder.Append("class ");
+                    }
+                    else if (cciTypeParameter.MustBeValueType)
+                    {
+                        builder.Append("valuetype ");
+                    }
+
+                    if (cciTypeParameter.AllowsRefLikeType)
+                    {
+                        builder.Append("byreflike ");
+                    }
+
+                    if (cciTypeParameter.MustHaveDefaultConstructor)
+                    {
+                        builder.Append(".ctor ");
+                    }
+
+                    appendTypeParameterTypeConstraints(typeParameter, builder);
+                }
+                else
+                {
+                    builder.Append('!');
+                }
+
+                builder.Append('T');
+                builder.Append(StringExtensions.GetNumeral(typeParameter.Ordinal));
+            }
+
+            static void appendTypeParameterTypeConstraints(TypeParameterSymbol typeParameter, StringBuilder builder)
+            {
+                ImmutableArray<TypeWithAnnotations> typeConstraints = typeParameter.GetConstraintTypes(ConsList<TypeParameterSymbol>.Empty);
+                if (typeConstraints.IsEmpty && !typeParameter.HasUnmanagedTypeConstraint && !typeParameter.HasValueTypeConstraint)
+                {
+                    return;
+                }
+
+                var typeConstraintStrings = ArrayBuilder<string>.GetInstance(typeConstraints.Length);
+                foreach (var typeConstraint in typeConstraints)
+                {
+                    var constraintBuilder = PooledStringBuilder.GetInstance();
+                    appendType(typeConstraint.Type, constraintBuilder.Builder);
+                    typeConstraintStrings.Add(constraintBuilder.ToStringAndFree());
+                }
+
+                if (typeParameter.HasUnmanagedTypeConstraint)
+                {
+                    typeConstraintStrings.Add("class System.ValueType modreq(class System.Runtime.InteropServices.UnmanagedType)");
+                }
+                else if (typeParameter.HasValueTypeConstraint)
+                {
+                    typeConstraintStrings.Add("class System.ValueType");
+                }
+
+                typeConstraintStrings.Sort(StringComparer.Ordinal); // Actual order doesn't matter - just want to be deterministic
+
+                builder.Append('(');
+                for (int i = 0; i < typeConstraintStrings.Count(); i++)
+                {
+                    if (i > 0)
```

Comment:
>if (i > 0) [](http://example.com/codeflow?start=20&length=10)

For simplicity, we could always add comma at the end of each type as we add spaces after non-type constraints.

Reply:
I kept this as-is. The trailing comma just feels ugly and the complexity is minimal